docs/patches.md

```diff
@@ -0,0 +1,60 @@
+# Snapfu Patch Files
+This document describes how to leverage various features available within a Snapfu patch file. Snapfu patch files can be used to:
+* run commands
+* modify files
+
+## YAML
+Each patch descriptor file is a YAML file that describes the changes to be made to a project. These files must be parsable by a YAML parser. A simple change to package dependencies can be seen below:
+```yaml
+version: 0.40.0
+description: 'updating snap dependencies to version 0.40.0'
+
+steps:
+    - files:
+        package.json:
+            action: edit-json
+            changes:
+                - update:
+                    properties: {
+                        dependencies: {
+                            "@searchspring/snap-preact": "^0.40.0",
+                            "@searchspring/snap-preact-components": "^0.40.0",
+                        }
+                    }
+```
+
+## Version
+The version property details which version the patch is part of. Valid patch versions are:  
+`[0-9].[0-9].[0-9]`
+
+## Description
+The description describes the changes being made.
+
+## Steps
+These are the steps to take when applying the patch. Any number of steps can be defined and are run sequentially.
+
+### Step type `run`
+Various commands can be executed using the `run` step. In most cases this command would be used to copy over new files, remove existing files or rename files. The commands used here expect a Unix environment of some sort. Commonly utilized commands might include the `cp`, `mv` and `rm` commands.
```

Comment:
What do you think about adding a `run` to the example above or here?

Reply:
Example added.